### ISSUE TITLE
mutual information gradient estimation

### ISSUE THREAD

**Issue by ghost** [oozdenizci/MMIDimReduction#1]
Thank you for your excellent work, but I have a question. In Figure 2 the mutual information gradient is used as a loss function to optimise the parameters of the network. Is the mutual information between Y and C maximum when the value of the loss function of the mutual information gradient is lower?

**Comment by oozdenizci** [comment]
Hi - Thanks a lot for your question! I assume you refer to the Figure 2 from the paper "Stochastic mutual information gradient estimation for dimensionality reduction networks". I'll try to clarify here, but I also suggest following the story-line in the manuscript rather than Fig.2 alone.

The mutual information gradient is not used "as" the loss function in optimization. Here, the instantaneous estimate of mutual information is incorporated in stochastic gradient descent for optimization (the figure aims to illustrate SMIG being used as the gradient of an approximated loss).

Basically the idea was the following: Projected output y is a function of parameters of the network (i.e., theta), and one should update the parameters of the network via SGD such that I(Y,C) can be maximized (Eq. 3). This however necessitates the true gradient of the mutual information \nabla I(Y,C), which is not easy to obtain. Hence we stochastically approximate this true gradient of the mutual information, by performing SGD parameter updates using an instantaneous estimate of this gradient, \nabla I_t(Y,C), which we can estimate non-parameterically. I hope this clarifies better!

**Comment by ghost** [comment]
Thank you for your reply! I read your article again and it helped me a lot. I choose to close this issue.
